### ISSUE TITLE
update test cases to LLVM 11.0

### ISSUE THREAD

**Issue by mewmew** [llir/llvm#158]
The test cases have been updated to LLVM 11.0, using the same steps as in https://github.com/llir/llvm/issues/132#issue-588625688

Use branch [llvm-11.0-testdata](https://github.com/llir/testdata/tree/llvm-11.0-testdata) of the testdata repository.

These are the currently failing test cases.

# Test case mismatch

Note, it is quite likely that a few of the original test cases have been updated. In these cases, we must also update our .golden file of those test cases.

For errors like `asm_test.go:531: module "../testdata/llvm/test/Analysis/CostModel/AMDGPU/fdiv.ll.golden" mismatch (-want +got):`

See https://github.com/llir/llvm/issues/132#issuecomment-604634673 for example on how to update these kind of test cases.

* Golden test case has to be updated
	- [x] `testdata/llvm/test/Analysis/CostModel/AMDGPU/fdiv.ll.golden`
	- [x] `testdata/llvm/test/Transforms/InstCombine/fma.ll.golden`
	- [x] `testdata/llvm/test/Transforms/InstCombine/vec_demanded_elts.ll.golden`
	- [x] `testdata/llvm/test/Assembler/auto_upgrade_intrinsics.ll.golden`
	- [x] `testdata/llvm/test/Assembler/fast-math-flags.ll.golden`
	- [x] `testdata/llvm/test/DebugInfo/X86/DIModuleContext.ll.golden`
	- [x] `testdata/llvm/test/Transforms/InstCombine/vec_demanded_elts.ll.golden`

Fixed in rev llir/testdata@a3f3b7a2aad988afa25ce068721393babd7b3736.

# Parse errors

These requires update to the LLVM IR grammar (for inspiration, see #101 for update of grammar to LLVM 9.0)

For errors like `asm_test.go:516: unable to parse "../testdata/llvm/test/Transforms/InstSimplify/compare.ll" into AST; syntax error at line 1676`:

* Update grammar
	- [x] `testdata/llvm/test/Assembler/dimodule.ll`
	- `DICompileUnit` (`rangesBaseAddress`, `sysroot`)
		- [x] [`testdata/llvm/test/Assembler/dicompileunit.ll`](https://github.com/llir/testdata/blob/a3f3b7a2aad988afa25ce068721393babd7b3736/llvm/test/Assembler/dicompileunit.ll#L26) (line 26)
		- [x] [`testdata/llvm/test/DebugInfo/X86/DIModule.ll`](https://github.com/llir/testdata/blob/a3f3b7a2aad988afa25ce068721393babd7b3736/llvm/test/DebugInfo/X86/DIModule.ll#L21) (line 21)
		- [x] [`testdata/llvm/test/DebugInfo/X86/clang-module.ll`](https://github.com/llir/testdata/blob/a3f3b7a2aad988afa25ce068721393babd7b3736/llvm/test/DebugInfo/X86/clang-module.ll#L23) (line 23)
	- `null_pointer_is_valid`
		- [x] [`testdata/llvm/test/Transforms/Attributor/nonnull.ll`](https://github.com/llir/testdata/blob/a3f3b7a2aad988afa25ce068721393babd7b3736/llvm/test/Transforms/Attributor/nonnull.ll#L1234) (line 1234)
		- [x] [`testdata/llvm/test/Transforms/InstSimplify/compare.ll`](https://github.com/llir/testdata/blob/a3f3b7a2aad988afa25ce068721393babd7b3736/llvm/test/Transforms/InstSimplify/compare.ll#L1676) (line 1676)

**Comment by dannypsnl** [comment]
- testdata/llvm/test/Transforms/InstSimplify/compare.ll:
    `panic: unable to parse integer "u0x100000001"; strconv.ParseInt: parsing "u0x100000001": invalid syntax`

**Comment by dannypsnl** [comment]
And
- testdata/llvm/test/DebugInfo/X86/DIModule.ll
- testdata/llvm/test/DebugInfo/X86/clang-module.ll
seems stuck by DIModule parsing error, DIModule sysroot cannot be parsed(but it should).

**Comment by mewmew** [comment]
It seems the `sysroot` value is not quoted in the output.

Current output of `llir/llvm`:
```llvm
!0 = distinct !DICompileUnit(language: DW_LANG_ObjC, file: !1, producer: "LLVM version 3.7.0", runtimeVersion: 2, emissionKind: FullDebug, enums: !2, retainedTypes: !2, globals: !2, imports: !3, sysroot: /)
```

Should be `sysroot: "/"`

**Comment by mewmew** [comment]
This issue has now been resolved, and all LLVM 11.0 test cases are passing.
